acceptance/internal/config.go

```diff
@@ -149,3 +152,56 @@ func DoLoadConfig(t *testing.T, path string) TestConfig {
 
 	return config
 }
+
+// This function takes EnvMatrix and expands into a slice of environment configurations.
+// Each environment configuration is a slice of env vars in standard Golang format.
+// For example,
+//
+//	input: {"KEY": ["A", "B"], "OTHER": ["VALUE"]}
+//
+// output: [["KEY=A", "OTHER=VALUE"], ["KEY=B", "OTHER=VALUE"]]
+//
+// If any entries is an empty list, that variable is dropped from the matrix before processing.
+func ExpandEnvMatrix(matrix map[string][]string) [][]string {
+	result := [][]string{{}}
+
+	if len(matrix) == 0 {
+		return result
+	}
+
+	// Filter out keys with empty value slices
+	filteredMatrix := make(map[string][]string)
+	for key, values := range matrix {
+		if len(values) > 0 {
+			filteredMatrix[key] = values
+		}
+	}
+
+	if len(filteredMatrix) == 0 {
+		return result
+	}
+
+	keys := make([]string, 0, len(filteredMatrix))
+	for key := range filteredMatrix {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	for _, key := range keys {
```

Comment:
This section is really hard to read. Can you simplify this? Or at least add doc strings to explain the ideas behind each important line of code?

Reply:
Added a comment explaining the algorithm here. I suggest to also look at test cases to understand what's happening.